_overviews/overview/scala-features.md

```diff
@@ -0,0 +1,496 @@
+
+* [Generic classes](/tour/generic-classes.html)
+* [Variance annotations](/tour/variances.html)
+* [Upper](/tour/upper-type-bounds.html) and [lower](/tour/lower-type-bounds.html) type bounds
+* [Inner classes](/tour/inner-classes.html) and [abstract type members](/tour/abstract-type-members.html) as object members
+* [Compound types](/tour/compound-types.html)
+* [Explicitly typed self references](/tour/self-types.html)
+* [Implicit parameters](/tour/implicit-parameters.html) and [conversions](/tour/implicit-conversions.html)
+* [Polymorphic methods](/tour/polymorphic-methods.html)
+* [Type inference](/tour/type-inference.html) means the user is not required to annotate code with redundant type information. 
+
+In combination, these features provide a powerful basis for the safe reuse of programming abstractions and for the type-safe extension of software.
+
+<!--
+https://www.artima.com/scalazine/articles/scalas_type_system.html
+MartinOdersky:
+The first is that the type system in Scala is actually very flexible, so it typically lets you compose things in very flexible patterns, which a language like Java, which has a less expressive type system, would often make more difficult. The second is that with pattern matching, you can recover type information in a very flexible way without even noticing it.
+
+MartinOdersky (https://jaxenter.com/current-state-scala-odersky-interview-129495.html):
+I think one of the next big developments will be the refinement of type systems to describe ever more precise properties of programs.
+
+https://en.wikipedia.org/wiki/Scala_(programming_language)#Expressive_type_system
+- Expressive type system
+  - Classes and abstract types as object members
+  - Structural types
+  - Path-dependent types
+  - Compound types
+  - Explicitly typed self references
+  - Generic classes
+  - Polymorphic methods
+  - Upper and lower type bounds
+  - Variance (covariance and contravariance)
+  - Annotation
+  - Views
+  - Type inference
+  - “Type enrichment” (implicits)
+  - Statically typed
+  - ADTs
+
+- Possible resources
+    - https://en.wikipedia.org/wiki/Scala_(programming_language)#Expressive_type_system
+    - https://docs.scala-lang.org/tour/tour-of-scala.html
+    - https://www.artima.com/scalazine/articles/scalas_type_system.html
+    - https://heather.miller.am/blog/types-in-scala.html
+
+- DOT, Soundness
+    - http://dotty.epfl.ch/blog/2016/02/17/scaling-dot-soundness
+    - https://www.scala-lang.org/blog/2016/02/03/essence-of-scala.html
+-->
+
+
+### It’s a pure functional programming language
+
+Scala is a functional programming (FP) language, meaning:
+
+- Functions are variables, and can be passed around like any other variable
+- Higher-order functions are directly supported
+- Lambdas are built in
+- Everything in Scala is an expression that returns a value
```

Comment:
Technically speaking "everything" is not true. For example `class A` is not an expression. This is only the case for statements.